android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketModule.java

```diff
@@ -213,23 +220,83 @@ public void removeListeners(Integer count) {
         // Keep: Required for RN built in Event Emitter Calls.
     }
 
-    private void requestNetwork(final int transportType) throws InterruptedException {
+    private void requestNetwork(final int transportType, @Nullable final String iotDeviceHost) throws InterruptedException {
         final NetworkRequest.Builder requestBuilder = new NetworkRequest.Builder();
         requestBuilder.addTransportType(transportType);
         final CountDownLatch awaitingNetwork = new CountDownLatch(1); // only needs to be counted down once to release waiting threads
         final ConnectivityManager cm = (ConnectivityManager) mReactContext.getSystemService(Context.CONNECTIVITY_SERVICE);
-        cm.requestNetwork(requestBuilder.build(), new ConnectivityManager.NetworkCallback() {
-            @Override
-            public void onAvailable(Network network) {
-                currentNetwork.setNetwork(network);
-                awaitingNetwork.countDown(); // Stop waiting
-            }
 
-            @Override
-            public void onUnavailable() {
+        // smartmedev - add support for for concurrent-connections:
+        // Route all data to the ioT device network interface if exist more than one concurrent network
+        // See: https://developer.android.com/about/versions/12/behavior-changes-12#concurrent-connections
+        if (cm != null) {
+            // Get all connected networks
+            Network[] allNetworks = cm.getAllNetworks();
+            List<Network> wifiNetworks = new ArrayList<>();
+
+            // Check exist at least one newtwork
+            if (allNetworks != null && allNetworks.length > 0) {
+                // Filter for retreive only networks based on selected transport type
+                for (Network network : allNetworks) {
+                    NetworkCapabilities nc = cm.getNetworkCapabilities(network);
+                    if (nc != null && nc.hasTransport(transportType)) {
+                        wifiNetworks.add(network);
+                    }
+                }
+
+                // Check exist at least one newtwork based on selected transport type
+                if (!wifiNetworks.isEmpty()) {
+                    boolean networkFound = false;
+                    for (Network network : wifiNetworks) {
+                        LinkProperties linkProperties = cm.getLinkProperties(network);
+                        // Ensure linkProperties is not null
+                        if (linkProperties == null)
+                            continue;
+
+                        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
+                            Inet4Address foundServerAddress = linkProperties.getDhcpServerAddress();
+                            if(iotDeviceHost.equals(foundServerAddress.getHostAddress())) {
```

Comment:
what happens if iotDeviceHost is null ? I cannot see any check before.

Reply:
Hi @vricosti iotDeviceHost can never be null, accordingly to main library readme.md you should see that if "host" is not specified in the configuration, it would assume "localhost" value.

This is the "connect" method of Socket.js file:
`
connect(options, callback) {
        const customOptions = { ...options };
        // Normalize args
        customOptions.host = customOptions.host || 'localhost';
        customOptions.port = Number(customOptions.port) || 0;
        this.once('connect', () => {
            if (callback) callback();
        });
        this._connecting = true;
        this._readyState = 'opening';
        Sockets.connect(this._id, customOptions.host, customOptions.port, customOptions);
        return this;
    }
`

I was thinking that "host" param in configuration array is mandatory, isn't it?
I can add a new check to use old behavior if iotDeviceHost.equals("localhost") the default value.

Let me know if is a good idea

Reply:
Ok I see but in this case why iotDeviceHost is @Nullable ?
For the rest I am not respnsible for this project  but I assume it would be good to keep old behavior except if your implementation is better.
Now I hope that your branch will be merged because the project don't seem to be very active.

Reply:
Yes I think the same.
I pushed a new commit on master branch that added old behavior if "host" not specified in configuration array.
So the library take the old behavior if iotDeviceHost is null or equals "localhost" -> default value for "host" param.